xknx/devices/climate_mode.py

```diff
@@ -218,55 +253,108 @@ async def _set_internal_operation_mode(self, operation_mode):
+
     @property
     def operation_modes(self):
         """Return all configured operation modes."""
         if not self.supports_operation_mode:
             return []
-        return self.operation_modes_
+        return self._operation_modes
+
+    @property
+    def controller_modes(self):
+        """Return all configures controller modes."""
+        if not self.supports_controller_mode:
+            return []
+        return self._controller_modes
 
     def gather_operation_modes(self):
         """Gather operation modes from RemoteValues."""
         operation_modes = []
-        for rv in self._iter_remote_values():
+        for rv in self._iter_binary_remote_values():
+            if rv.writable:
+                operation_modes.extend(rv.supported_operation_modes())
+
+        for rv in self._iter_normal_operation_modes():
+            if rv.writable:
+                operation_modes.extend(rv.supported_operation_modes())
+
+        # remove duplicates
+        return list(set(operation_modes))
+
+    def gather_controller_modes(self):
+        """Gather controller modes from RemoteValues."""
+        operation_modes = []
+        for rv in self._iter_controller_remote_values():
             if rv.writable:
                 operation_modes.extend(rv.supported_operation_modes())
+
         # remove duplicates
         return list(set(operation_modes))
 
     async def process_group_write(self, telegram):
         """Process incoming and outgoing GROUP WRITE telegram."""
         if self.supports_operation_mode:
-            processed = False
             for rv in self._iter_remote_values():
                 if await rv.process(telegram):
-                    # don't set when binary climate mode rv is False
                     if rv.value:
                         await self._set_internal_operation_mode(rv.value)
-                        return
-                    processed = True
-            # if no operation mode has been set and all binary operation modes are False
-            if processed:
-                await self._set_internal_operation_mode(HVACOperationMode.STANDBY)
+
+        if self.supports_controller_mode:
+            for rv in self._iter_controller_remote_values():
+                if await rv.process(telegram):
+                    if rv.value:
+                        await self._set_internal_controller_mode(rv.value)
 
     async def sync(self, wait_for_result=False):
         """Read states of device from KNX bus."""
         if self.supports_operation_mode:
-            for rv in self._iter_remote_values():
+            for rv in chain(
+                self._iter_normal_operation_modes(), self._iter_binary_remote_values()
+            ):
+                await rv.read_state(wait_for_result=wait_for_result)
+
+        if self.supports_controller_mode:
+            for rv in self._iter_controller_remote_values():
                 await rv.read_state(wait_for_result=wait_for_result)
```

Comment:
supports_operation_mode and supports_controller_mode checks for .initialized so I think it would be save to just iterate through all RVs - rv.read_state checks for .readable first.

We could then even remove the whole sync() function to use the super()-implementation from RemoteValue.